x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/retention/ExpiredForecastsRemover.java

```diff
@@ -79,7 +84,8 @@ public void remove(ActionListener<Boolean> listener) {
 
         SearchRequest searchRequest = new SearchRequest(RESULTS_INDEX_PATTERN);
         searchRequest.source(source);
-        client.execute(SearchAction.INSTANCE, searchRequest, forecastStatsHandler);
+        client.execute(SearchAction.INSTANCE, searchRequest, new ThreadedActionListener<>(LOGGER, threadPool,
+                MachineLearning.UTILITY_THREAD_POOL_NAME, forecastStatsHandler, false));
```

Comment:
I think the `removeDataBefore()` method in `ExpiredModelSnapshotsRemover` should also use a `ThreadedActionListener` in exactly the same way this class does.  It also has the problem of doing a (potentially) large amount of parsing on the network thread.

Reply:
Good point. I pushed a commit to fix that too.